### PR TITLE
Fix clamp broadcasting on MPS (Fixes #160734)

### LINKED ISSUE #160734: MPS clamp operation does not fully support broadcasting
### 🐛 Describe the bug

The CPU clamp operation fully supports broadcasting between the input and the min/max tensors. However MPS fails when the number of dimensions of min or max is greater than that of input tensor

```python
import torch

def test_clamp(device):
    x = torch.randn(2, 3, device=device)
    min_t = torch.randn(1, 2, 3, dtype=x.dtype, device=device)
    max_t = torch.randn(1, 2, 3, dtype=x.dtype, device=device)
    try:
        torch.clamp(x, min=min_t, max=max_t)
        print(f"clamp test succeeds for device: {device}")
    except RuntimeError as e:
        print(f"clamp test fails for device: {device}: {e}")

test_clamp(device = "cpu")
test_clamp(device = "mps")
```

terminal output:
```
clamp test succeeds for device: cpu
clamp test fails for device: mps: impl: clamp tensor number of dims must not be greater than that of input tensor
```

### Versions

nightly

cc @malfet @kulinseth @albanD @DenisVieriu97 @jhavukainen

### PR DESCRIPTION
**Pull request by roei-shlezinger**
This PR fixes a bug where `torch.clamp` on MPS fails when min/max tensors have more dimensions than the input tensor. 
CPU already supports this broadcasting, but MPS raised a RuntimeError.

Example of failing case before the fix:
```python
x = torch.randn(2, 3, device="mps")
min_t = torch.randn(1, 2, 3, device="mps")
max_t = torch.randn(1, 2, 3, device="mps")
torch.clamp(x, min=min_t, max=max_t)  # RuntimeError
```
After this fix, MPS matches CPU behavior.

Fixes #160734

### CODE REVIEW

**Comment by roei-shlezinger** [comment]
Hi! I’m seeing two CI failures that don’t seem related to my changes: the DTensor dropout test on CPU, and the lintrunner job stopping because conda isn’t available in the linter image. lintrunner -a passes locally for me. @kulinseth could you advise on the next steps? Thanks!

**Review by malfet — APPROVED** [review]
